onehealth_db/postgresql_database.py

```diff
@@ -1128,71 +1114,22 @@ def get_var_values_nuts(
 
     # get the variable type id
     var_id = var_type.id
-    # now query all variable values within the NUTS regions for this time point
-    # get variable values for each grid point and time point
-    query = (
-        session.query(
-            VarValue.value.label("var_value"),
-            GridPoint.point.label("geometry"),
-            TimePoint.year,
-            TimePoint.month,
-            TimePoint.day,
-            VarType.name.label("var_name"),
-        )
-        .join(GridPoint, VarValue.grid_id == GridPoint.id)
-        .join(TimePoint, VarValue.time_id == TimePoint.id)
-        .join(VarType, VarValue.var_id == VarType.id)
-        .filter(
-            GridPoint.id.in_(grid_ids_in_nuts),
-            TimePoint.id == time_id,
-            VarType.id == var_id,
-        )
-    )
-    compiled_query = query.statement.compile(
-        dialect=postgresql.dialect(), compile_kwargs={"literal_binds": True}
-    )
-    var_values = gpd.read_postgis(
-        compiled_query,
-        engine,
-        params={
-            "grid_ids_in_nuts": list(grid_ids_in_nuts),
-            "time_id": time_id,
-            "var_id": var_id,
-        },
-        geom_col="geometry",
+
+    # now query all variable values with their nuts_id for this time point
+    values = (
+        session.query(VarValueNuts)
+        .filter(VarValueNuts.time_id == time_id, VarValueNuts.var_id == var_id)
+        .all()
     )
-    if var_values.empty:
-        print("No variable values found for the specified criteria.")
+    # now get all the nuts ids associated with the values
+    nuts_ids = [v.nuts_id for v in values]
+    if not nuts_ids:
+        print("No NUTS id's found in the database.")
         raise HTTPException(
-            status_code=400, detail="No variable values found in the database."
-        )
-    # aggregate variable values for each NUTS region
-    aggregated_by_nuts = (
-        gpd.sjoin(
-            var_values,
-            nuts_regions,
-            how="inner",
-            predicate="intersects",
+            status_code=400, detail="No NUTS ids found in the database."
         )
-        .groupby(["nuts_id", "var_name", "time"])
-        .agg(
-            {
-                "var_value": "mean",  # average value for the variable
-            }
-        )
-        .reset_index()
-    )
-    # merge the aggregated values with the NUTS regions
-    # create a list of tuples with (NUTS_ID, var_value)
-    nuts_var_values = nuts_regions.merge(aggregated_by_nuts, on="nuts_id")
-
-    mydict = {"NUTS id, var_value": nuts_var_values}
-    print(mydict)
-    with open("nuts_grid_data_onehealth.json", "w") as f:
-        import json
-
-        json.dump(mydict, f, indent=4)
-        print("Data saved to cartesian_grid_data_onehealth.json")
+    # create a dict with NUTS_id: var_value
+    mydict = {nuts_id: v.value for v, nuts_id in zip(values, nuts_ids)}
     return mydict
```

Comment:
two questions for this function:
- this should at least return an empty dict, if no nuts_id: value pair can be placed (should not be possible due to the exceptions, but if, it should still be a dict and not `None`)
- I opted for returning a dict with `nuts_id: value`, it is of course also possible to return a tuple `(nuts_id, value)`; I thought the dict should result in less overhead for you @jmsssc - is that the case, or would you prefer a different data structure?